### PR TITLE
Remove sensitive fields when reading config data

### PR DESCRIPTION
**Pull request by calvn**

### CODE REVIEW

File: builtin/credential/ldap/path_config.go [hunk 1]
```diff
@@ -319,7 +321,7 @@ type ConfigEntry struct {
 	InsecureTLS   bool   `json:"insecure_tls" structs:"insecure_tls" mapstructure:"insecure_tls"`
 	StartTLS      bool   `json:"starttls" structs:"starttls" mapstructure:"starttls"`
 	BindDN        string `json:"binddn" structs:"binddn" mapstructure:"binddn"`
-	BindPassword  string `json:"bindpass" structs:"bindpass" mapstructure:"bindpass"`
+	BindPassword  string `json:"bindpass" structs:"bindpass,omitempty" mapstructure:"bindpass"`
```
Comment (jefferai) [inline builtin/credential/ldap/path_config.go]: Better idea: remove `structs` entirely...

File: builtin/credential/radius/path_config.go [hunk 2]
```diff
@@ -199,7 +201,7 @@ func (b *backend) pathConfigCreateUpdate(ctx context.Context, req *logical.Reque
 type ConfigEntry struct {
 	Host                     string   `json:"host" structs:"host" mapstructure:"host"`
 	Port                     int      `json:"port" structs:"port" mapstructure:"port"`
-	Secret                   string   `json:"secret" structs:"secret" mapstructure:"secret"`
+	Secret                   string   `json:"secret" structs:"secret,omitempty" mapstructure:"secret"`
```
Comment (jefferai) [inline builtin/credential/radius/path_config.go]: Same here.

File: builtin/logical/cassandra/backend.go [hunk 3]
```diff
@@ -66,11 +66,11 @@ type backend struct {
 type sessionConfig struct {
 	Hosts           string `json:"hosts" structs:"hosts" mapstructure:"hosts"`
 	Username        string `json:"username" structs:"username" mapstructure:"username"`
-	Password        string `json:"password" structs:"password" mapstructure:"password"`
+	Password        string `json:"password" structs:"password,omitempty" mapstructure:"password"`
```
Comment (jefferai) [inline builtin/logical/cassandra/backend.go]: And here

File: builtin/logical/mongodb/path_config_connection.go [hunk 4]
```diff
@@ -47,6 +47,8 @@ func (b *backend) pathConnectionRead(ctx context.Context, req *logical.Request,
 	if err := entry.DecodeJSON(&config); err != nil {
 		return nil, err
 	}
+	config.URI = ""
+
```
Comment (jefferai) [inline builtin/logical/mongodb/path_config_connection.go]: And here

File: builtin/logical/mssql/path_config_connection.go [hunk 5]
```diff
@@ -53,6 +53,8 @@ func (b *backend) pathConnectionRead(ctx context.Context, req *logical.Request,
 	if err := entry.DecodeJSON(&config); err != nil {
 		return nil, err
 	}
+	config.ConnectionString = ""
+
```
Comment (jefferai) [inline builtin/logical/mssql/path_config_connection.go]: And here

File: builtin/logical/mysql/path_config_connection.go [hunk 6]
```diff
@@ -63,6 +63,8 @@ func (b *backend) pathConnectionRead(ctx context.Context, req *logical.Request,
 	if err := entry.DecodeJSON(&config); err != nil {
 		return nil, err
 	}
+	config.ConnectionURL = ""
+
```
Comment (jefferai) [inline builtin/logical/mysql/path_config_connection.go]: And here

File: builtin/logical/postgresql/path_config_connection.go [hunk 7]
```diff
@@ -73,6 +73,8 @@ func (b *backend) pathConnectionRead(ctx context.Context, req *logical.Request,
 	if err := entry.DecodeJSON(&config); err != nil {
 		return nil, err
 	}
+	config.ConnectionURL = ""
+
```
Comment (jefferai) [inline builtin/logical/postgresql/path_config_connection.go]: And here

**Comment by jefferai** [comment]
One meta comment: in a lot of places we are adding warnings to the write endpoint indicating that the response will contain the password in cleartext. Since that's no longer the case we should remove those.

**Comment by jefferai** [comment]
I think you missed radius.

**Comment by jefferai** [comment]
Nope, I'm wrong

**Comment by calvn** [comment]
I had a feeling you would bring up `structs` 😄.

**Pull request merged**